include/swift/AST/DiagnosticsSema.def

```diff
@@ -3043,9 +3043,12 @@ ERROR(transpose_attr_overload_not_found,none,
 ERROR(transpose_attr_cannot_use_named_wrt_params,none,
       "cannot use named 'wrt' parameters in '@transpose(of:)' attribute, found "
       "%0", (Identifier))
-ERROR(transpose_attr_result_value_not_differentiable,none,
-      "'@transpose(of:)' attribute requires original function result %0 to "
-      "conform to 'Differentiable'", (Type))
+ERROR(transpose_func_needs_static,none,
+     "the transpose of an instance method needs to be static when "
+     "differentiating wrt 'self'.", ())
+ERROR(transpose_func_self_static_types_not_match,none,
```

Comment:
`transpose_func_self_static_types_not_match` is subsumed by `transpose_func_wrt_self_must_be_static`, but `transpose_func_wrt_self_must_be_static` doesn't show the type mismatch information.

We can replace `transpose_func_self_static_types_not_match` with a note that shows the type mismatch. The note would be emitted right after `transpose_func_wrt_self_must_be_static`.

```suggestion
NOTE(transpose_func_wrt_self_self_type_mismatch_note,none,
     "the transpose is declared in %0 but the original function is declared in "
     "%1", (Type, Type))
```